modules/rostests/apitests/shell32/ShellExecuteEx.cpp

```diff
@@ -111,16 +111,37 @@ getCommandLineFromProcess(HANDLE hProcess)
     PEB peb;
     PROCESS_BASIC_INFORMATION info;
     RTL_USER_PROCESS_PARAMETERS Params;
+    NTSTATUS Status;
+    BOOL ret;
+
+    Status = NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+
+    ret = ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
-    NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
-    ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
     ReadProcessMemory(hProcess, peb.ProcessParameters, &Params, sizeof(Params), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
     LPWSTR cmdline = Params.CommandLine.Buffer;
-    SIZE_T cchCmdLine = Params.CommandLine.Length;
-    LPWSTR pszBuffer = (LPWSTR)calloc(cchCmdLine + 1, sizeof(WCHAR));
-    ReadProcessMemory(hProcess, cmdline, pszBuffer, cchCmdLine, NULL);
-    pszBuffer[cchCmdLine] = UNICODE_NULL;
+    if (!cmdline)
+        trace("!cmdline\n");
+
+    SIZE_T cbCmdLine = Params.CommandLine.Length;
+    if (!cbCmdLine)
+        trace("!cbCmdLine\n");
+
+    LPWSTR pszBuffer = (LPWSTR)calloc(cbCmdLine + sizeof(WCHAR), sizeof(WCHAR));
```

Comment:
Calloc syntax is: `calloc(number_of_elements, size_of_one_single_element)`.
So here you're uselessly allocating (twice) more memory than needed, since your number of elements is already a number of bytes, and therefore the size of one single "element" is the size of one byte (in bytes), which is 1 .
Thus the call should be `calloc(cbCmdLine + sizeof(WCHAR), 1);`

Reply:
```suggestion
    LPWSTR pszBuffer = (LPWSTR)calloc(cbCmdLine + sizeof(WCHAR), 1);
```